packages/browser-agent-core/src/common/wrap/wrap-history.js

```diff
@@ -3,30 +3,45 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-// History pushState wrapper
-import {ee as baseEE} from '../event-emitter/contextual-ee'
-import {createWrapperWithEmitter as wfn} from './wrap-function'
+/**
+ * Wrapper for pushState and replaceState methods of window.history object
+ */
+
+import { ee as globalEE } from '../event-emitter/contextual-ee'
+import { createWrapperWithEmitter as wfn } from './wrap-function'
 import { isBrowserScope } from '../util/global-scope'
 
-const wrapped = {}
+let alreadyWrapped = false
 
+/**
+ * Wraps the `pushState` and `replaceState` methods of `window.history` and returns a corresponding event emitter
+ * scoped to the history object.
+ * @param {Object} sharedEE - The shared event emitter.
+ * @returns {Object} Scoped event emitter with a debug ID of `history`.
+ */
 export function wrapHistory(sharedEE){
   const ee = scopedEE(sharedEE)
-  if (wrapped[ee.debugId] || !isBrowserScope) return ee; // History API is only relevant within web env
-  wrapped[ee.debugId] = true
+  if (alreadyWrapped || !isBrowserScope) return ee; // History API is only relevant within web environment.
+  alreadyWrapped = true
   var wrapFn = wfn(ee)
 
-  var prototype = window.history && window.history.constructor && window.history.constructor.prototype
-  var object = window.history
-  if (prototype && prototype.pushState && prototype.replaceState) {
-    object = prototype
-  }
-
-  wrapFn.inPlace(object, [ 'pushState', 'replaceState' ], '-')
+  /**
+   * For objects that will be instantiated more than once, we wrap the object's prototype methods. The history object
+   * is instantiated only once, so we can wrap its methods directly--and we must wrap the history methods directly as
+   * long as [Chromium issue 783382](https://bugs.chromium.org/p/chromium/issues/detail?id=783382) remains unresolved.
+   */
+  wrapFn.inPlace(window.history, [ 'pushState', 'replaceState' ], '-')
```

Comment:
Maybe change this to `wrapFn.inPlace(globalScope.history`. `globalScope` should point to `window` here but the globalScope variable was created so we dont have window all over our code base.

Reply:
Just want to clarify that window was changed to `self` to accommodate workers without too much duplication/redundancy. And then `self` broke because of that ticket w/ customer redefining it. 
The point: keep `window` as `window` where relevant, and use `globalScope` when it's something shared with worker build/agent function. In this case, there's no `history` API in worker, so it doesn't quite make sense to ask `window or workerGlobalScope ? window.history : workerGlobalScope.history` @bjfield